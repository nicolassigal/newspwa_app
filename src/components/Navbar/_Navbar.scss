.navbar {
    height: 50px;
    display: flex;
    flex-direction: row;
    align-items: center;
    flex-wrap: wrap;
    z-index: 1;
    background: var(--bg-card-color);
    padding: 30px 20px;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    border-bottom: 1px solid rgba(0,0,0,0.1);
    ul {
        padding:0;
        margin:0;
    }
    .brand {
        margin:0;
        font-size: 24px;
        padding-right: 20px;
        margin-right: 20px;
        border-right: 2px solid var(--txt-color);
        order: 1;
        h1 {
            margin:0;
            padding:0;
            color: var(--txt-color);
        }
    }

    #menuToggle {
        order: 2;
    }

    input, span {
        display: none;
    }
    a {
        padding-right: 20px;
        text-decoration: none;
        color: var(--txt-color);
        &:active,
        &:hover
        &:focus,
        &:active:focus {
            text-decoration: none;
            color: var(--txt-color);
        }
        &.active {
            color: $gold;
        }
    }
}

@media (max-width: 1024px)  {
    a {
    text-decoration: none;
    color: var(--txt-color);
    transition: color 0.3s ease;
    font-size: 24px;
    padding: 15px 0px;
    }

    .navbar {
        display: flex;
        justify-content: flex-start;
        align-items: center;
        padding:20px;
        .brand {
            font-size: 18px;
            border: none;
            padding: 0;
            margin: 0;
            order: 2;
            padding-left: 20px;
            h1 {
                margin:0;
                padding:0;
                color: var(--txt-color);
            }
        }
    }

    #menuToggle {
    display: block;
    position: relative;
    top: 0;
    left: 0;
    z-index: 1;
    order: 1;
    -webkit-user-select: none;
    user-select: none;
    input {
        display: block;
        width: 40px;
        height: 32px;
        position: absolute;
        top: -7px;
        left: -5px;
        
        cursor: pointer;
        
        opacity: 0;
        z-index: 2;
        
        -webkit-touch-callout: none;
        &:checked ~ ul {
            transform: none;
        }
        &:checked ~ span {
            opacity: 1;
            transform: rotate(45deg) translate(-2px, -1px);
            background: #232323;
            &:nth-last-child(2) {
                transform: rotate(-45deg) translate(0, -1px);
            }
            &:nth-last-child(3) {
                opacity: 0;
                transform: rotate(0deg) scale(0.2, 0.2);
            }
        }
    }
    span {
        display: block;
        width: 33px;
        height: 4px;
        margin-bottom: 5px;
        position: relative;
        
        background: #cdcdcd;
        border-radius: 3px;
        
        z-index: 1;
        
        transform-origin: 4px 0px;
        
        transition: transform 0.5s cubic-bezier(0.77,0.2,0.05,1.0),
                    background 0.5s cubic-bezier(0.77,0.2,0.05,1.0),
                    opacity 0.55s ease;

        &:first-child {
            transform-origin: 0% 0%;
        }

        &:nth-last-child(2) {
            transform-origin: 0% 100%;
        }
    }
    }

    #menu {
    position: absolute;
    width: 300px;
    margin: -100px 0 0 -50px;
    padding: 50px;
    padding-top: 125px;
    display: flex;
    flex-direction: column;
    background: var(--sidenav-bg);
    list-style-type: none;
    -webkit-font-smoothing: antialiased;
    
    transform-origin: 0% 0%;
    transform: translate(-100%, 0);
    
    transition: transform 0.5s cubic-bezier(0.77,0.2,0.05,1.0);
    width: 50vw;
    height: 100vh;
    & li {
        padding: 10px 0;
        font-size: 22px;
    }
    }
}