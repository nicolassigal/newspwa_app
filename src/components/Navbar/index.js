import React from 'react';
import { NavLink } from 'react-router-dom';
import NightMode from '../NightMode';

const Navbar = (props) => {
    return (
        <nav className="navbar">
        <div id="menuToggle">
            <input type="checkbox" />
            <span></span>
            <span></span>
            <span></span>
            <ul id="menu">
                <NavLink exact={true} activeClassName="active" to="/">Home</NavLink>
                {
                    props.categories.map((category, i) => (
                        <NavLink 
                        key={i}
                        activeClassName="active"
                        to={{pathname: `/categories/${category}`, query: { category }}}>
                        {category}
                        </NavLink>
                    ))
                }
            </ul>
            </div>
            <div className="brand">
                <h1>News</h1>
            </div>
            <NightMode />
        </nav>
    )
}

export default Navbar;