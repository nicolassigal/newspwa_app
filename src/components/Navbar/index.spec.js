import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Navbar from './index';
import { shallow } from 'enzyme';

describe('Navbar component test', () => {
  it('Navbar component should render correctly', () => {
    const props = {
      categories: []  
    }


   const wrapper = shallow(<Navbar {...props}/>)
   expect(wrapper).toMatchSnapshot();

  });

  it('Navbar component should render with list', () => {
    const props = {
        categories: [
           "cat_1",
           "cat_2"
        ]  
    }


   const wrapper = shallow(<Navbar {...props}/>)
   expect(wrapper).toMatchSnapshot();

   const Articles = wrapper.find("#menu").getElement();
   expect(Articles.props.children).toHaveLength(2);
  });
});