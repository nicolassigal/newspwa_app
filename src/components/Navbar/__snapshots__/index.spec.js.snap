// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component test Navbar component should render correctly 1`] = `ShallowWrapper {}`;

exports[`Navbar component test Navbar component should render with list 1`] = `ShallowWrapper {}`;
